### PR TITLE
[Microsoft Defender for Endpoint]: Fix bug be able to use different operators in filter 

### PR DESCRIPTION
**Pull request by pl-brault**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
When using the command `microsoft-atp-list-machine-actions-details` the string representing the filter that will be pass to the api are made with only AND and eq operator (using the fonction `reformat_filter`).

To be able to filter on the field `creationDateTimeUtc` we need different operator like  ge, le, gt, lt. To avoid changing multiple fonctions, I've added the `filters` argument to be able to pass custom filter made by the user. It gives the possibility to filter on the field `creationDateTimeUtc` and the possibility to use the OR operator in the string representing the filter that will be pass to the api.

Here is an example of the command:
`!microsoft-atp-list-machine-actions-details filters="(type eq 'Isolate' or type eq 'Unisolate') and creationDateTimeUtc ge 2023-10-20T12:00:57.1234Z"`

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 1]
```diff
@@ -2943,7 +2944,7 @@ def get_machine_action_by_id_command(client: MsClient, args: dict):
             'type': type,
             'requestor': requestor
         }
-        filter_req = reformat_filter_with_list_arg(fields_to_filter_by, "machineId")
+        filter_req = filters if filters else reformat_filter_with_list_arg(fields_to_filter_by, "machineId")
```
Comment (MosheEichler) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py:2947]:
```suggestion
        filter_req = filters or reformat_filter_with_list_arg(fields_to_filter_by, "machineId")
```

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.yml [hunk 2]
```diff
@@ -1284,6 +1284,8 @@ script:
     - deprecated: true
       description: Flag for the rate limit retry.
       name: ran_once_flag
+    - name: filters
+      description: String representation of filters.
```
Comment (MosheEichler) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.yml:1288]:
```suggestion
      description: String representation of filters (Override every other filters).
```

File: Packs/MicrosoftDefenderAdvancedThreatProtection/ReleaseNotes/1_16_16.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#### Integrations
+##### Microsoft Defender for Endpoint
+- Added *filters* argument for the command `microsoft-atp-list-machine-actions-details`.
```
Comment (MosheEichler) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/ReleaseNotes/1_16_16.md:3]:
```suggestion
- Added the **filters** argument to the ***microsoft-atp-list-machine-actions-details*** command.
- Updated the Docker image to: *demisto/crypto:1.0.0.79610*.
```

**Review by MosheEichler — COMMENTED** [review]
Hi @pl-brault,
Thank you for your contribution!
Good work :)

Plesae see my comments.

Please feel free to reach out to me with any questions - I'm available here or on slack :)
Thanks again

**Comment by ShirleyDenkberg** [comment]
@MosheEichler Doc review completed.

**Review by MosheEichler — APPROVED** [review]

**Pull request merged**